src/CareTogether.Core/Engines/PolicyEvaluation/ReferralCalculations.cs

```diff
@@ -9,6 +9,36 @@
 
 namespace CareTogether.Engines.PolicyEvaluation
 {
+    using System;
+    using System.Collections.Immutable;
+    using System.Text;
+
+    public static class ImmutableListExtensions
+    {
+        public static string ToHumanFriendlyString<T>(this ImmutableList<T> list)
+        {
+            if (list == null || list.Count == 0)
```

Comment:
This method can be made more concise and readable using [the `String.Join` method](https://learn.microsoft.com/en-us/dotnet/api/system.string.join?view=net-8.0#system-string-join(system-string-system-collections-generic-ienumerable((system-string)))) and C# string interpolation:

```c#
public static string ToHumanFriendlyString<T>(this ImmutableList<T> list)
{
    if (list == null || !list.Any())
    {
        return "The list is empty.";
    }

    return $"List contents: [{string.Join(", ", list)}]";
}
```